### ISSUE TITLE
[YSQL] ORDER BY condition on an indexed column does not order rows correctly.

### ISSUE THREAD

**Issue by m-iancu** [YugaByte/yugabyte-db#979]
To replicate:
```
postgres=# create table test(a int, b int);
CREATE TABLE
postgres=# create index test_b on test(b);
CREATE INDEX
postgres=# insert into test(a,b) values (1,1), (2,2), (3,3), (4,4), (5,5) ;
postgres=# select * from test order by b DESC;
 a | b
---+---
 5 | 5
 1 | 1
 4 | 4
 2 | 2
 3 | 3
(5 rows)
```
Ordering by non-index columns works correctly. E.g: 
```
postgres=# select * from test order by a DESC;
 a | b
---+---
 5 | 5
 4 | 4
 3 | 3
 2 | 2
 1 | 1
(5 rows)
```

**Comment by robertpang** [comment]
This happens because YugaByte indexes piggyback on PostgreSQL's btree index access methods.  PG's btree indexes support full ordering whereas YugaByte indexes doesn't currently support full ordering yet.

To fix the ordering issue, we should mark YugaByte indexes' with `canorder = false` in the access method declaration until full ordering is supported.

**Comment by robertpang** [comment]
Similarly, YugaByte indexes return empty bitmap scan causing incorrect result:

```
postgres=# create table test (v1 int, v2 int, v3 int);
CREATE TABLE
postgres=# create index on test (v2, v3);
CREATE INDEX
postgres=# insert into test values (1, 11, 21), (2, 12, 22), (3, 13, 23), (4, 14, 24), (5, 15, 25);
INSERT 0 5
postgres=# delete from test where v2 = 12 or v2 = 13 returning v2;
 v2
----
(0 rows)
postgres=# explain delete from test where v2 = 12 or v2 = 13;
                                          QUERY PLAN
-----------------------------------------------------------------------------------------------
 Delete on test  (cost=0.50..4.51 rows=1000 width=68)
   ->  Bitmap Heap Scan on test  (cost=0.50..4.51 rows=1000 width=68)
         Recheck Cond: ((v2 = 12) OR (v2 = 13))
         ->  BitmapOr  (cost=0.50..0.50 rows=1 width=0)
               ->  Bitmap Index Scan on test_v2_v3_idx  (cost=0.00..0.00 rows=1 width=0)
                     Index Cond: (v2 = 12)
               ->  Bitmap Index Scan on test_v2_v3_idx  (cost=0.00..0.00 rows=1 width=0)
                     Index Cond: (v2 = 13)
(8 rows)

**Comment by robertpang** [comment]
Resolved by commit [f702d72](https://github.com/YugaByte/yugabyte-db/commit/f702d726e9de8981e263b57a40ef89e49566e677).
